course_info/templates/course_info/partials/people.html

```diff
@@ -38,263 +43,250 @@ <h3 class="breadcrumbs">
+<div ng-if="messages.success" class="row add-top-margin">
+  <div class="col-xs-12">
+    <div class="panel panel-default" id="alert-pane">
+      <div class="panel-body" id="alert-pane">
 
-  <!-- begin add partial failure warning alerts -->
-  <uib-alert ng-if="addPartialFailure" type="warning"
-             close="closeAlert('addPartialFailure')" class="col-md-12">
-    <p>While adding <strong>{{ addPartialFailure.searchTerm }}</strong> to this course:</p>
-    <p>{{ addPartialFailure.text }}</p>
-  </uib-alert>
-  <!-- end add partial failure warning alerts -->
+        <!-- begin success alerts -->
+        <uib-alert ng-if="messages.success" type="{{messages.success.alertType}}"
+                   id="alert-success">
+          <span ng-if="messages.success.type == 'add'">
+            <ng-pluralize count="tracking.successes"
+                          when="{'0': 'No people were added to the course.',
+                                 'one': '1 person was added to the course.',
+                                 'other': '{} people were added to the course.'}">
+            </ng-pluralize>
+            <ng-pluralize count="tracking.total_failures"
+                          when="{'0': '',
+                                 'one': '1 person could not be added.',
+                                 'other': '{} people could not be added.'}">
+            </ng-pluralize>
+          </span>
+          <span ng-if="messages.success.type == 'remove'">
+            <p><strong>{{ messages.success.searchTerm }}</strong>
+              was just removed from this course:</p>
+            <div class="course-membership-entry">
+              {{ getProfileFullName(messages.success.profile) }}
+              <badge role="{{ getProfileRoleTypeCd(messages.success.profile) }}"></badge>
+              {{ messages.success.user_id }}
+              <strong>Enrolled as {{ messages.success.role.role_name }}</strong>
+            </div>
+          </span>
+        </uib-alert>
+        <!-- end success alerts -->
 
-  <!-- begin remove failure warning alerts -->
-  <uib-alert ng-if="removeFailure" type="warning"
-             close="closeAlert('removeFailure')" class="col-md-12">
-    <div ng-switch="removeFailure.type">
-      <div ng-switch-when="noSuchUser">
-        <strong>{{ getProfileFullName(removeFailure.profile) || removeFailure.user_id }}</strong>
-        has already been removed from this course.
-      </div>
-      <div ng-switch-when="noSuchCourse">
-        <strong>{{ getProfileFullName(removeFailure.profile) || removeFailure.user_id }}</strong>
-        cannot be removed from this course because it no longer exists.
-      </div>
-      <div ng-switch-when="serverError">
-        There was a problem removing
-        <strong>{{ getProfileFullName(removeFailure.profile) || removeFailure.user_id }}</strong>
-        from the database. Please try again.
+        <div ng-if="warningsToDisplay()" id="table-failure-details">
```

Comment:
Oh, wait, I remember.  `ng-if` on a table gets you horrible flickering.  Never mind.